src/components/DevPanel.tsx

```diff
@@ -0,0 +1,369 @@
+      body: formData,
+    }).then((response) => response.json())
+      .then((responseJSON) => {
+        const {
+          status,
+        } = JSON.parse(responseJSON);
+        if (status === 'SUCCESS') {
+          alert.show(`Successfully uploaded ${pdfFile.name}`);
+          this.setState({
+            buttonState: '',
+            pdfFiles: undefined,
+          }, () => this.getDocuments());
+        } else {
+          alert.show(`Failure to upload ${pdfFile.name}`);
+          this.setState({ buttonState: '' });
+        }
+      });
+  }
+
+  handleChangeFileDownload(event: any, rowIndex: number) {
+    event.preventDefault();
+    const {
+      alert,
+    } = this.props;
+    const { files } = event.target;
+
+    this.setState({
+      pdfFiles: files,
+    }, () => this.handleFileDownload(rowIndex));
+  }
+
+  handleFileDownload(rowIndex: number) {
+    const {
+      userRole,
+    } = this.props;
+
+    const documentId = this.state.documents[rowIndex].id;
+    const documentName = this.state.documents[rowIndex].filename;
+
+    let pdfType;
+    if (userRole === Role.Worker || userRole === Role.Admin || userRole === Role.Director) {
+      pdfType = PDFType.FORM;
+    } else if (userRole === Role.Client) {
+      pdfType = PDFType.IDENTIFICATION;
+    } else {
+      pdfType = undefined;
+    }
+
+    fetch(`${getServerURL()}/download`, {
+      method: 'POST',
+      credentials: 'include',
+      body: JSON.stringify({
+        fileId: documentId,
+        pdfType,
+      }),
+    }).then((response) => response.blob())
+      .then((response) => {
+        const pdfFile = new File([response], documentName, { type: 'application/pdf' });
+        console.log(pdfFile);
+
+        const url = window.URL.createObjectURL(response);
+        const a = document.createElement('a');
+        a.href = url;
+        a.download = documentName;
+        document.body.appendChild(a);
+        a.click();
+        a.remove();
+      }).catch((error) => {
+        alert('Error Fetching File');
+      });
+  }
+
+  ButtonFormatter = (cell, row, rowIndex, formatExtraData) => (
+    <div>
+      <label className="btn btn-filestack btn-widget ml-5 mr-5">
+        Re-Upload
+        <input type="file" accept="application/pdf" id="potentialPdf" multiple onChange={(event) => this.handleChangeFileUpload(event, rowIndex)} hidden />
+      </label>
+      <label className="btn btn-filestack btn-widget ml-5 mr-5">
+        Download
+        <button onClick={(event) => this.handleChangeFileDownload(event, rowIndex)} hidden />
+      </label>
+    </div>
+  )
+
+  tableCols = [{
+    dataField: 'filename',
+    text: 'Application Name',
+    sort: true,
+  }, {
+    dataField: 'category',
+    text: 'Category',
+    sort: true,
+  }, {
+    text: 'Actions',
+    formatter: this.ButtonFormatter,
+  }];
+
+  constructor(props: Props) {
+    super(props);
+    this.handleChangeFileUpload = this.handleChangeFileUpload.bind(this);
```

Comment:
can you use arrow functions so you dont have to bind them: 
e.g.
handleChangeFileUpload = () => {}

Reply:
This is a design change we need to do in all the code. I will not do it for now, since the other code has it